extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/AbstractJpaOperations.java

```diff
@@ -414,7 +420,21 @@ public boolean exists(Class<?> entityClass, String query, Parameters params) {
     }
 
     public long deleteAll(Class<?> entityClass) {
+        Metamodel metamodel = getEntityManager().getMetamodel();
+        EntityType<?> entity1 = metamodel.entity(entityClass);
+        Set<Attribute<?, ?>> declaredAttributes = ((EntityTypeImpl) entity1).getDeclaredAttributes();
+        if (declaredAttributes.stream().filter(attribute -> attribute.getPersistentAttributeType()
+                .equals(Attribute.PersistentAttributeType.ELEMENT_COLLECTION)).count() > 0) {
```

Comment:
Is this enough? Don't you also need to consider associations with cascade delete enabled?

Reply:
Unfortunately it looks like JPA doesn’t expose the cascade style so you would need to look at something Hibernate-specific.

Reply:
Yes, effectively. I need to detect the both cases. I will push very soon my code re-factored. Thank you for the proposition.